meshtastic/module_config.proto

```diff
@@ -173,10 +192,9 @@ message ModuleConfig {
     uint32 monitor_pin = 6;
 
     /*
-     * Whether or not the GPIO pin state detection is triggered on HIGH (1)
-     * Otherwise LOW (0)
+     * The type of trigger event to be used
      */
-    bool detection_triggered_high = 7;
+    TriggerType detection_trigger_type = 7;
```

Comment:
Looks good. I think we also want a max_size 8 for the underlying int in the options file to make sure we don't alloc a full 32bit uint for this field.